Update compile_to_js.sh to explicite export fucntions

marwin1991 opened the pull request:
When compile with WASM=0 functions are not exported. I reported it here kripken/emscripten#7721
I think for now we can stay with this explicite function and we can remove unused in WEB to make compiled library lighter.

grzanka commented on distributions/JavaScript/compile_to_js.sh:
```diff
@@ -21,8 +21,457 @@ ls .
 emcmake cmake ..
 emmake make
 
+funs='['	
```
Do we have a script which generates list of functions, as below ?
If yes, add a comment line mentioning how to do it

marwin1991 replied:
It is described in JS_INSTALL.txt file

grzanka replied:
ok

grzanka commented:
Why tests are failing ? On master they run without problems. Is it a "feature" of forjs branch ? Can you adjust travis for this branch ?

marwin1991 commented:
```
0.09s$ cd distributions/Python/pyamtrack && python generate_Py_wrapper.py
Reading NAMESPACE file
Harvesting function definitions in header files
Traceback (most recent call last):
  File "generate_Py_wrapper.py", line 46, in <module>
    main()
  File "generate_Py_wrapper.py", line 40, in main
    c_functions = py_parser.harvest(desired_func)
  File "/home/travis/build/libamtrack/library/distributions/Python/pyamtrack/py_parser.py", line 250, in harvest
    elif string.split(string.split(tmp_header_content[tmp_definition_start])[1], '(') [0] in functions:              
TypeError: list indices must be integers, not type
The command "cd distributions/Python/pyamtrack && python generate_Py_wrapper.py" exited with 1.
0.05s$ if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then LD_LIBRARY_PATH=$HOME/usr/lib python pyamtrack.py; fi
python: can't open file 'pyamtrack.py': [Errno 2] No such file or directory
The command "if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then LD_LIBRARY_PATH=$HOME/usr/lib python pyamtrack.py; fi" exited with 2.
```

Why this is failing? Can you look at it ?

grzanka commented:
@marwin1991 lack of doxygen comments in include/AT_CernlibFuns.h ?

grzanka commented on .travis.yml:
```diff
@@ -34,4 +34,6 @@ script:
\ No newline at end of file
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then LD_LIBRARY_PATH=$HOME/usr/lib python pyamtrack.py; fi  # FIX IT TO WORK WITH OSX
+  # Tigger rebuild image in docker hub libamtrack/libamtrackforjs
+  - curl -X POST https://cloud.docker.com/api/build/v1/source/f382631a-cc27-4630-9c15-79598e025ea0/trigger/65027d52-ffaf-46f4-8006-96614edda072/call/
```
Is URL completely transparent ? Shouldn't it be more secured to avoid random people triggering docker hub rebuild ?

marwin1991 replied:
Done

marwin1991 commented:
@grzanka Do you think we can merge this to forjs branch?
Forjs branch will be waitng till we will merge Landau and Vavilov C implementation to master and then merge from master to forjs branch and see if everything works

grzanka commented:
It is OK now, please merge.
